vm/src/vm.rs

```diff
@@ -227,175 +227,169 @@ impl<ENV> GVm<ENV> {
 
     pub fn is_equal_pair(&self, val1: Value, val2: Value) -> VMResult<Value> {
         let mut val = Value::False;
-        if val1 == val2 {
-            val = Value::True;
-        } else if val1.is_int() && val2.is_int() {
-            if val1.get_int(self)? == val2.get_int(self)? {
-                val = Value::True;
+        match (val1, val2) {
+            (_, _) if val1 == val2 => val = Value::True,
+            (Value::Int(i1), Value::Int(i2)) if from_i56(&i1) == from_i56(&i2) => val = Value::True,
```

Comment:
I think this is handled by (_, _).  Any of the basic (non-heap, non-float types) when comparing to the same type should be covered by (_, _) I think.